bool smooth
---
float32[] plan
bool success
